lib/web3EventService.ts

```diff
@@ -296,27 +213,289 @@ export class Web3EventService {
+   * by name and Arc.js contract wrapper.
+   * @param requiredDepth -- If set then will not invoke the callback until the transaction has been mined to
+   * the requiredDepth.Pass -1 to use the Arc.js's global default depth.  Note that the Fetcher
+   */
+  public aggregatedEventsFetcherFactory(
+    events: Array<EventToAggregate>,
+    requiredDepth: number = 0)
+    : EntityFetcherFactory<AggregatedEventsResult, Hash> {
+
+    if (!events) {
+      throw new Event("events was not supplied");
+    }
+    /**
+     * TODO:
+     * Rather than tracking transactions directly, track the mining of blocks and
+     * enumerate the transactions within them. This may give us the ability to go back
+     * in time, and will increase the chances of being able to aggregate events, that span multiple
+     * transactions, into a single event (when they all occur in a single block).
+     */
+    const baseFetcher = this.createFilterFetcherFactory("pending");
+
+    return this.createEntityFetcherFactory<AggregatedEventsResult, Hash>(
+      baseFetcher,
+      async (txHash: Hash): Promise<AggregatedEventsResult | undefined> => {
+
+        const foundEvents = new Map<EventToAggregate, DecodedLogEntryEvent<any>>();
+
+        const txReceipt = await TransactionService.watchForMinedTransaction(
+          txHash,
+          undefined,
+          requiredDepth) as TransactionReceipt;
+
+        for (const log of txReceipt.logs) {
+
+          const contractAddress = log.address;
+          /**
+           * see if a contract of interest generated this log
+           */
+          for (const eventSpec of events) {
+
+            // IContractWrapperBase
+            const foundContract = (contractAddress === eventSpec.contract.address) ? eventSpec.contract : null;
+
+            if (foundContract) {
+              /**
+               * get the decoded events for this contract
+               */
+              (txReceipt as any).receipt = null; // force it to redo
+              const txReceiptDecoded = await TransactionService.toTxTruffle(txReceipt, foundContract.contract);
+              const decodedEvent =
+                txReceiptDecoded.logs.filter((l: DecodedLogEntryEvent<any>) => l.logIndex === log.logIndex)[0];
+              if (eventSpec.eventName === decodedEvent.event) {
+                foundEvents.set(eventSpec, decodedEvent);
+              }
+            }
+          }
+        }
+        if (foundEvents.size) {
+          return { events: foundEvents, txReceipt };
+        }
+      });
+  }
+
+  /**
+   * Convert the EntityFetcherFactory<TEntitySrc, TEntityOriginalSrc> into an
+   * EntityFetcherFactory<TEntityDest, TEntitySrc>.
+   *
+   * @param entityFetcherFactory The source EntityFetcherFactory
+   * @param transformEventCallback Converts TEntitySrc into TEntityDest
+   */
+  public pipeEntityFetcherFactory<TEntityDest, TEntitySrc, TEntityOriginalSrc>(
```

Comment:
What does T stand for at the beginning of the types?

Reply:
`T` stands for `Type`